packages/integrations/vercel/src/index.ts

```diff
@@ -130,6 +130,21 @@ function getAdapter({
 			i18nDomains: 'experimental',
 			envGetSecret: 'stable',
 		},
+		client: {
+			internalFetchHeaders: skewProtection
+				? (): Record<string, string> => {
+						const deploymentId = process.env.VERCEL_DEPLOYMENT_ID;
+						if (deploymentId) {
+							return { 'x-deployment-id': deploymentId };
+						}
+						return {};
+					}
+				: undefined,
+			assetQueryParams:
+				skewProtection && process.env.VERCEL_DEPLOYMENT_ID
```

Comment:
How come the query string logic checks for `skewProtection` boolean but the headers logic above doesn't?

Also, can we make the `skewProtection` boolean default to `process.env.VERCEL_SKEW_PROTECTION_ENABLED === '1'`?

See the docs https://vercel.com/docs/skew-protection#supported-frameworks